src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const result = [];
+
+  for (const item of actions) {
+    const stateCopy = { ...currentState };
+
+    if (item.type === 'addProperties') {
+      Object.assign(stateCopy, { ...item.extraData });
+    }
+
+    if (item.type === 'removeProperties') {
+      for (const el of item.keysToRemove) {
+        delete stateCopy[el];
+      }
+    }
+
+    if (item.type === 'clear') {
+      for (const param in stateCopy) {
+        delete stateCopy[param];
+      }
+    }
+
+    result.push(stateCopy);
+    currentState = stateCopy;
```

Comment:
The action of pushing the `stateCopy` into the `result` array is repeated in every case of the action type handling. Consider performing this operation only once after the switch statement to adhere to the DRY (Don't Repeat Yourself) principle.